src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -97,7 +97,7 @@ def getClassInfoForEnvironment(env)
     # and so would not be subject to any potentially stale settings data being
     # used to enumerate manifests.
     environment = @env_loader.list.find do |env_from_loader|
-      env_from_loader.name == env
+      env_from_loader.name.to_s == env
```

Comment:
`env_from_loader.name` is a `Symbol` but `env` is a `String`.  For reasons I don't understand, the original comparison passes under JRuby 1.7.X but fails on JRuby 9K - but seemingly unique to this code for some reason.

The JRuby 9K behavior seems correct to me - and is consistent with what I see when doing simple `Symbol` to `String` comparisons from `irb` under MRI Ruby 1.9.3 and 2.1 and `jirb` under JRuby 1.7.22 (under Ruby 1.9.3) and 9.0.4.0.  I also did this against code pulled from the master branch:

````
# puppetserver ruby --version
jruby 1.7.20.1 (1.9.3p551) 2015-06-10 d7c8c27 on OpenJDK 64-Bit Server VM 1.7.0_51-mockbuild_2014_06_09_19_19-b00 +indy +jit [linux-amd64]
# puppetserver ruby -e "puts :mysym == 'mysym'"
false
````